controller/internal/controller/consumer_controller.go

```diff
@@ -181,3 +209,74 @@ func (r *ConsumerReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		)
 	return controller.Complete(r)
 }
+
+// checkConsumerConflicts Perform full conflict detection and rebuild the index
+func (r *ConsumerReconciler) checkConsumerConflicts(ctx context.Context, state *consumerReconcileState) {
+	r.KeyIndex.mu.Lock()
+	defer r.KeyIndex.mu.Unlock()
+
+	if state == nil || state.namespaceToConsumers == nil {
+		return
+	}
+
+	// Clear old indexes
+	r.KeyIndex.index = make(map[string]map[string]map[string]string)
+
+	// Check conflicts for all valid consumers in the current state
+	for ns, consumers := range state.namespaceToConsumers {
+		// Create a new map to filter out invalid consumers
+		validConsumers := make(map[string]*mosniov1.Consumer)
+
+		for name, consumer := range consumers {
+			if err := r.indexConsumer(ns, consumer); err == nil {
+				validConsumers[name] = consumer
+			} else {
+				consumer.SetAccepted(mosniov1.ReasonInvalid, err.Error())
```

Comment:
BTW, we should also check the newly added conflict mark in the integration test.